js/reference.js

```diff
@@ -5825,3 +5843,75 @@
             return citation;
         }
     };
+
+    /**
+     * Constructs the Google Dataset metadata for the given tuple.
+     * The given metadata must be valid and have the appropriate variables.
+     */
+    function GoogleDatasetMetadata(reference, gdsMetadataAnnotation) {
+        this._reference = reference;
+        this._table = reference.table;
+        this._gdsMetadataAnnotation = gdsMetadataAnnotation;
+    }
+
+    GoogleDatasetMetadata.prototype = {
+        /**
+         * Given the templateVariables variables, will generate the metadata.
+         * @param {ERMrest.Tuple} tuple - the tuple object that this metadata is based on
+         * @param {Object=} templateVariables - if it's not an object, we will use the tuple templateVariables
+         * @return {Json-ld|null} if the returned template for required attributes are empty or invalid, it will return null.
+         */
+        compute: function (tuple, templateVariables) {
+            var table = this._table,
+                metadataAnnotation = this._gdsMetadataAnnotation;
+
+            if (!templateVariables) {
+                templateVariables = tuple.templateVariables.values;
+            }
+
+            var metadata = {};
+            setMetadataFromTemplate(metadata, metadataAnnotation.dataset, metadataAnnotation.template_engine, templateVariables, table);
+
+            // remove null attributes so they don't get included in the json
+            metadata = removeEmptyOrNull(metadata);
```

Comment:
Why do you have to do this before calling the validator? Why shouldn't this be part of the validator given that it's already making some other modifications to the input?